data-prepper-plugins/split-event-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/splitevent/SplitEventProcessor.java

```diff
@@ -0,0 +1,127 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.processor.splitevent;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.DefaultEventHandle;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+
+@DataPrepperPlugin(name = "split_event", pluginType = Processor.class, pluginConfigurationType = SplitEventProcessorConfig.class)
+public class SplitEventProcessor extends AbstractProcessor<Record<Event>, Record<Event>>{
+    final Pattern pattern;
+    final String delimiter;
+    final String delimiterRegex;
+    final String field;
+
+    @DataPrepperPluginConstructor
+    public SplitEventProcessor(final PluginMetrics pluginMetrics, final SplitEventProcessorConfig config) {
+        super(pluginMetrics);
+        this.delimiter = config.getDelimiter();
+        this.delimiterRegex = config.getDelimiterRegex();
+        this.field = config.getField();
+
+        if(delimiterRegex != null && !delimiterRegex.isEmpty()
+                && delimiter != null && !delimiter.isEmpty()) {
+            throw new IllegalArgumentException("delimiter and delimiter_regex cannot be defined at the same time");
+        } else if((delimiterRegex == null || delimiterRegex.isEmpty()) &&
+                (delimiter == null || delimiter.isEmpty())) {
+            throw new IllegalArgumentException("delimiter or delimiter_regex needs to be defined");
+        }
+
+        if(delimiterRegex != null && !delimiterRegex.isEmpty()) {
+            pattern = Pattern.compile(delimiterRegex);
+        } else {
+            pattern = Pattern.compile(Pattern.quote(delimiter));
+        }
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        Collection<Record<Event>> newRecords = new ArrayList<>();
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+
+            if (!recordEvent.containsKey(field)) {
+                Record newRecord = new Record<>(recordEvent);
+                newRecords.add(newRecord);
+                continue;
+            }
+            
+            final Object value = recordEvent.get(field, Object.class);
+
+            //split record according to delimiter
+            final String[] splitValues = pattern.split((String) value);
```

Comment:
Rather than create a `Pattern` for both regex and delimiter, I think you could make a `Function`. Then you can avoid a regex.

```
private final Function<String, String[]> splitter;
...
@DataPrepperPluginConstructor
public SplitEventProcessor(final PluginMetrics pluginMetrics, final SplitEventProcessorConfig config) {
...
        if(delimiterRegex != null && !delimiterRegex.isEmpty()) {
            Pattern pattern = Pattern.compile(delimiterRegex);
            splitter = inputString -> pattern.split(inputString);
        } else {
            splitter = inputString -> inputString.split(delimiter);
        }
}
```

Then this line becomes:

```
final String[] splitValues = splitter.apply((String) value);
```